include/seqan3/argument_parser/validators.hpp

```diff
@@ -322,16 +323,16 @@ class file_ext_validator // TODO AFTER INITIAL MERGE: Check if file exists, also
         extensions{v}
     {}
 
-    /*!\brief Tests whether cmp lies inside extensions.
-     * \param cmp The input value to check.
+    /*!\brief Tests whether path lies inside extensions.
+     * \param path The input value to check.
      * \throws parser_invalid_argument
      */
-    void operator()(std::string const & cmp) const
+    void operator()(filesystem::path const & path) const
     {
-        std::string ext{cmp.substr(cmp.find_last_of('.') + 1)};
-
-        if (!(std::find(extensions.begin(), extensions.end(), ext) != extensions.end()))
-            throw parser_invalid_argument(detail::to_string("Extension ", ext, " is not one of ",
+      std::string cmp = path.string();
+      std::string ext{cmp.substr(cmp.find_last_of('.') + 1)};
```

Comment:
Having a filesystem path now, we can use it's member function [`filesystem::path::extension()`](https://en.cppreference.com/w/cpp/filesystem/path/extension) to get the extension. This ensures a correct extension even for some cases that we did not cover yet (e.g. `./` is a valid directory path but does not have a file extension. Our substr() command right now takes `/` as the extension while the fileysytem returns an empty extension).
Unfortunately `path.extension()` returns the extension WITH the dot so we need to drop that:
```
std::string ext{path.extension().string()};
ext = ext.substring(std::min(1, ext.size())); // drop '.' if extension is non-empty
```